pybamm/models/submodels/interface/open_circuit_potential/wycisk_ocp.py

```diff
@@ -0,0 +1,154 @@
+            variables[f"{Domain} electrode {phase_name}equilibrium OCP [V]"] = (
+                ocp_surf_eq
+            )
+
+            T_bulk = pybamm.xyz_average(pybamm.size_average(T))
+            ocp_bulk_eq = self.phase_param.U(sto_bulk, T_bulk)
+            variables[f"{Domain} electrode {phase_name}bulk equilibrium OCP [V]"] = (
+                ocp_bulk_eq
+            )
+
+            H = self.phase_param.hysteresis(sto_surf)
+            variables[f"{Domain} electrode {phase_name}OCP hysteresis [V]"] = H
+
+            dU = self.phase_param.U(sto_surf, T_bulk).diff(sto_surf)
+            dQ = self.phase_param.Q(sto_surf).diff(sto_surf)
+            dQdU = dQ / dU
+            variables[
+                f"{Domain} electrode {phase_name}differential capacity [A.s.V-1]"
+            ] = dQdU
+
+            H_x_av = pybamm.x_average(H)
+            h_x_av = pybamm.x_average(h)
+            variables[f"X-averaged {domain} electrode {phase_name}hysteresis state"] = (
+                h_x_av
+            )
+
+            # check if psd
+            if domain_options["particle size"] == "distribution":
+                # should always be true
+                if f"{domain} particle size" in sto_surf.domains["primary"]:
+                    # check if MPM Model
+                    if "current collector" in sto_surf.domains["secondary"]:
+                        ocp_surf = ocp_surf_eq + H_x_av * h_x_av
+                    # must be DFN with PSD model
+                    elif f"{domain} electrode" in sto_surf.domains["secondary"]:
+                        ocp_surf = ocp_surf_eq + H * h
+                elif (
+                    f"{domain} {phase_name}particle size" in sto_surf.domains["primary"]
+                ):
+                    ocp_surf = ocp_surf_eq + H * h
+            # must not be a psd
+            else:
+                ocp_surf = ocp_surf_eq + H * h
+
+            H_s_av = pybamm.size_average(H_x_av)
+            h_s_av = pybamm.size_average(h_x_av)
+
+            ocp_bulk = ocp_bulk_eq + H_s_av * h_s_av
+
+            dUdT = self.phase_param.dUdT(sto_surf)
+
+        variables.update(self._get_standard_ocp_variables(ocp_surf, ocp_bulk, dUdT))
+        return variables
+
+    def set_rhs(self, variables):
+        domain, Domain = self.domain_Domain
+        phase_name = self.phase_name
+
+        current = variables[
+            f"{Domain} electrode {phase_name}interfacial current density [A.m-2]"
+        ]
+        # check if composite or not
+        if phase_name != "":
+            Q_cell = variables[f"{Domain} electrode {phase_name}phase capacity [A.h]"]
+        else:
+            Q_cell = variables[f"{Domain} electrode capacity [A.h]"]
+
+        dQdU = variables[
+            f"{Domain} electrode {phase_name}differential capacity [A.s.V-1]"
+        ]
+        dQdU = dQdU.orphans[0]
+        K = self.phase_param.hysteresis_decay
+        K_x = self.phase_param.hysteresis_switch
```

Comment:
@valentinsulzer Do you want me to update the shorthand variable names within the OCP class, or are these fine to leave as is?

Reply:
Local within the OCP class is fine. As an aside, there is an argument to be made for defining the `FunctionParameter` and `Parameter` objects inside the specific submodels when they are only relevant for a single submodel. But it's not what we do for other submodels so let's leave it like this for now